src/context/src/asset_manager.rs

```diff
@@ -221,35 +220,21 @@ impl DerefMut for AssetManager {
     }
 }
 
-/// An enum with variants representing concrete
-/// `Factory` types compatible with different backends.
-pub enum FactoryImpl {
-    OpenGL { factory: gfx_device_gl::Factory },
-    #[cfg(windows)]
-    Direct3D {
-        // stub
-    },
-    Null,
-}
-
 impl AssetLoader<Mesh> for Vec<VertexPosNormal> {
     /// # Panics
     /// Panics if factory isn't registered as loader.
     fn from_data(assets: &mut Assets, data: Vec<VertexPosNormal>) -> Option<Mesh> {
```

Comment:
I'm not sure if the `AssetLoader` trait was added in this or the original context, but I feel that it should return a `Result<T, LoaderError>` or something similar. Given that data is unreliable (corrupt, parsed wrongly, etc.).